File: maui/KsDialogs.Maui.ApiSurfaceCheck/NegativeChecks/RejectsArgumentsOnNonePreset.cs
using System;

namespace KsDialogs.ApiSurfaceCheck.NegativeChecks;

/// <summary>none プリセットは引数を取らない (演出しないものに時間もイージングもない。core/ADR-0017)。</summary>
/// <remarks>
/// このソースはビルドプロパティ KsDialogsNegativeCheckNoneArguments を true にしたときだけ加わり、
/// <b>コンパイルエラーで失敗すること</b>が期待結果になる。
/// 期待する診断: CS1501 (引数 1 個を取る None のオーバーロードがない)
/// </remarks>
internal static class RejectsArgumentsOnNonePreset
{
    public static DialogTransition Build() => DialogTransition.None(TimeSpan.FromMilliseconds(200d));
}

File: android/api-surface-check/src/negativeCheckNoneArguments/kotlin/jp/kamusoft/ksdialogs/apicheck/RejectsArgumentsOnNonePreset.kt
package jp.kamusoft.ksdialogs.apicheck

import jp.kamusoft.ksdialogs.DialogTransition
import kotlin.time.Duration.Companion.milliseconds

/**
 * none プリセットは引数を取らない (演出しないものに時間もイージングもない。core/ADR-0017)。
 *
 * このソースは `-Pksdialogs.negativeCheck.noneArguments` を付けたときだけビルドに加わり、
 * **コンパイルエラーで失敗すること**が期待結果になる。
 */
public object RejectsArgumentsOnNonePreset {
    public fun build(): DialogTransition = DialogTransition.none(200.milliseconds)
}
